modules/internal/StringCasts.chpl

```diff
@@ -113,23 +113,7 @@ module StringCasts {
     var isErr: bool;
     // localize the string and remove leading and trailing whitespace
     var localX = x.localize();
-    const hasUnderscores = localX.find("_") != -1;
```

Comment:
I _think_ that `_cleanupStringForNumericCast` is missing the find optimizations (which tries to avoid any allocations if there are no underscores). I think you'll see a fairly large regression for `test/types/string/cassella/cast-from-string.chpl` https://github.com/chapel-lang/chapel/pull/11880 may have more context/info.